src/engine/SCons/Tool/__init__.py

```diff
@@ -913,15 +913,25 @@ def createCFileBuilders(env):
 #  Create common Java builders
 
 def CreateJarBuilder(env):
+    """The Jar builder expects a list of class files
+    which it can package into a jar file.
+
+    The jar tool provides an interface for passing other types
+    of java files such as .java, directories or swig interfaces
+    and will build them to class files in which it can package
+    into the jar.
+    """
     try:
-        java_jar = env['BUILDERS']['Jar']
+        java_jar = env['BUILDERS']['JarFile']
```

Comment:
Did you change the name of the builder here from Jar to JarFile?

Reply:
Yes, I added a builder wrapper in its place with the AddMethod function: 
https://github.com/SConsProject/scons/pull/17/commits/b54f4166d0053ae7168af1d16223fff574344f17#diff-06522ed2191b2caac07035c96d8cbfa4R199
I took this design from the Java builder with has a similar design pattern for taking in sources from both directories and java sources, see below to for comparison:
https://github.com/SConsProject/scons/blob/a583f043aec89e58bf4ab0ab20cc039299eff1df/src/engine/SCons/Tool/__init__.py#L954
https://github.com/SConsProject/scons/blob/a583f043aec89e58bf4ab0ab20cc039299eff1df/src/engine/SCons/Tool/javac.py#L163-L167